pytissueoptics/scene/tree/spacePartition.py

```diff
@@ -40,24 +41,33 @@ def searchPoint(self, point: Vector, node: Node = None) -> Node:
         isInside = None
         for child in node.children:
             if child.bbox.contains(point):
-                isInside = child
-                break
+                isInside = self.searchPoint(point, child)
 
         if isInside is None:
             if node.bbox.contains(point):
                 return node
 
-        return self.searchPoint(point, isInside)
+            else:
+                return None
+
+        return isInside
+
+    def findLeafNodesContainingPolygon(self, polygon):
+        pass
 
```

Comment:
What's up with that ?

Reply:
resolved with commit e9e8345717724422e1e353443aef9206531c3fd0